common/logisticspipes/blocks/powertile/LogisticsPowerJunctionTileEntity.java

```diff
@@ -31,33 +27,39 @@
 import logisticspipes.proxy.computers.interfaces.CCType;
 import logisticspipes.renderer.LogisticsHUDRenderer;
 import logisticspipes.utils.PlayerCollectionList;
-
 import net.minecraft.crash.CrashReportCategory;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.nbt.NBTTagCompound;
-import net.minecraft.tileentity.TileEntity;
-import net.minecraft.world.World;
-
 import net.minecraft.util.EnumFacing;
-
 import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.CapabilityInject;
 import net.minecraftforge.energy.CapabilityEnergy;
-import net.minecraftforge.energy.EnergyStorage;
 import net.minecraftforge.energy.IEnergyStorage;
 
-import ic2.api.energy.tile.IEnergySink;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+
+import static java.lang.Math.min;
 
-@ModDependentInterface(modId = { "IC2", "BuildCraft|Transport" }, interfacePath = { "ic2.api.energy.tile.IEnergySink", "buildcraft.api.power.IPowerReceptor" })
+@ModDependentInterface(modId = {"IC2"}, interfacePath = {"ic2.api.energy.tile.IEnergySink"})
 @CCType(name = "LogisticsPowerJunction")
 public class LogisticsPowerJunctionTileEntity extends LogisticsSolidTileEntity implements IGuiTileEntity, ILogisticsPowerProvider, IPowerLevelDisplay, IGuiOpenControler, IHeadUpDisplayBlockRendererProvider, IBlockWatchingHandler, IEnergySink {
 
 	public Object OPENPERIPHERAL_IGNORE; //Tell OpenPeripheral to ignore this class
 
+	@CapabilityInject(IMjConnector.class)
```

Comment:
I'm not 100% sure but I think this is going to crash if bc is not installed

Reply:
CapabilityInject doesn't do anything if there's no capability of that type registered

Reply:
But it tries to pass the Class and that class does not exist no CLassNotFoundException? Or am I missing a mechanic here?

Reply:
Forge does some magic with that, the ASMData thing gives you a string instead of a Class object when you ask it for the annotation value so it doesn't try to load the class.

Reply:
Ah OK, good to know.